Update CHANGELOG for version 1.83.0 (#10475)
treeverse/lakeFS: CHANGELOG.md

# Changelog

## v1.83.0

:new: What's new:

> - Introducing datasets - a new way to share curated data across your organization

Reviewer: ```suggestion
- Add UI entry points for the Enterprise-only Datasets feature.
```

:bug: Bugs Fixed:

> - Fixed a bug where async hooks could keep running after their triggering request was cancelled (#10454)

Reviewer: ```suggestion
- Ensure asynchronous hooks continue running after the triggering request is cancelled (#10454)
```
